modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/publisher/APIApplicationPublisherImpl.java

```diff
@@ -46,31 +51,36 @@ public void publish(APIApplication apiApplication) {
 				"APIApplicationPublisher not available");
 		}
 
-		String osgiJaxRsName = _getOSGiJaxRsName(apiApplication);
+		_endpointMatchersMap.put(
+			apiApplication.getBaseURL() + apiApplication.getCompanyId(),
```

Comment:
Mmmm, I think this could be dangerous if we accept a base url with numbers. What if someone creates a baseURL that ends with a number? We could be accessing to some protected information from other companies. Maybe we should include a separator